src/Sylius/Component/Shipping/Checker/Eligibility/CompositeShippingMethodEligibilityChecker.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Shipping\Checker\Eligibility;
+
+use Sylius\Component\Shipping\Model\ShippingMethodInterface;
+use Sylius\Component\Shipping\Model\ShippingSubjectInterface;
+use Webmozart\Assert\Assert;
+
+final class CompositeShippingMethodEligibilityChecker implements ShippingMethodEligibilityCheckerInterface
+{
+    /** @var ShippingMethodEligibilityCheckerInterface[] */
+    private $eligibilityCheckers;
+
+    /**
+     * @param ShippingMethodEligibilityCheckerInterface[] $eligibilityCheckers
+     */
+    public function __construct(array $eligibilityCheckers)
+    {
+        Assert::notEmpty($eligibilityCheckers);
```

Comment:
I would skip this assertion. If one does not register any checkers, this service would just not check anything 🚀